### PR TITLE
Filter out global team conversations for API v1 and v2.

### PR DESCRIPTION
**Pull request by elland**
Builds upon #2872 

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: services/galley/src/Galley/API/Query.hs [hunk 1]
```diff
@@ -415,11 +422,11 @@ conversationIdsPageFromV2 listGlobalSelf lusr Public.GetMultiTablePageRequest {.
     -- self-conversation.
     --
     -- FUTUREWORK: This is yet to be implemented for global team conversations.
```
Comment (smatting) [inline services/galley/src/Galley/API/Query.hs:424]: this FUTUREWORK can be removed

File: services/galley/test/integration/API/MLS.hs [hunk 2]
```diff
@@ -2210,6 +2211,42 @@ testGetGlobalTeamConv setup = do
...
+    let (Just gtc) = Aeson.decode rs :: Maybe GlobalTeamConversation
+        gid = cnvmlsGroupId $ gtcMlsMetadata gtc
+
+    void $ uploadNewKeyPackage alice1
+
+    -- create mls group
+    createGroup alice1 gid
+    void $ createAddCommit alice1 [] >>= sendAndConsumeCommitBundle
+
+  -- Now we should have the user as part of that conversation also in the backend
```
Comment (smatting) [inline services/galley/test/integration/API/MLS.hs:2245]: this comment should read differently, I think

File: services/galley/src/Galley/API/Query.hs [hunk 3]
```diff
@@ -358,34 +366,43 @@ conversationIdsPageFromV2 ::
   Sem r Public.ConvIdsPage
 conversationIdsPageFromV2 listGlobalSelf lusr Public.GetMultiTablePageRequest {..} = do
   let localDomain = tDomain lusr
+  gtcIds <-
+    E.getUserTeams (tUnqualified lusr) >>= mapM (pure . Data.globalTeamConv)
+
+  -- implicitly add user to the global team conv, so it gets listed.
+  for_ gtcIds $ \tid -> E.createMembers tid (UserList [tUnqualified lusr] mempty)
```
Comment (pcapriotti) [inline services/galley/src/Galley/API/Query.hs:389]: I thought the idea was that the GTC never has any explicit members.

File: services/galley/test/integration/API/MLS.hs (hunk 2 above)
Comment (elland) [reply services/galley/test/integration/API/MLS.hs]: Other way around, it should list for this user, since we're calling the right version of the API, no?

**Review by pcapriotti — CHANGES_REQUESTED** [review]
This doesn't look right to me. Since the normal listing function is not seeing the GTC, we can simply add it manually at the beginning (or the end) of the list we return, without the need for any special tricks.

**Pull request closed without merging**